packages/core/src/fhir-deduplication/resources/immunization.ts

```diff
@@ -76,53 +71,60 @@ export function groupSameImmunizations(immunizations: Immunization[]): {
     }
 
     const date = getDateFromResource(immunization, "datetime");
-    if (date && date !== "unknown") {
-      // TODO: should we keep date a mandatory field for dedup? If yes, then should we also add a default date to the FHIR encounter?
-      const { cvxCode, ndcCode } = extractCodes(immunization.vaccineCode);
-
-      if (cvxCode) {
-        const key = JSON.stringify({ date, cvxCode });
-        fillMaps(
-          immunizationsCvxMap,
-          key,
-          immunization,
-          refReplacementMap,
-          undefined,
-          assignMostDescriptiveStatus
-        );
-      } else if (ndcCode) {
-        const key = JSON.stringify({ date, ndcCode });
-        fillMaps(
-          immunizationsNdcMap,
-          key,
-          immunization,
-          refReplacementMap,
-          undefined,
-          assignMostDescriptiveStatus
-        );
-      } else {
-        const displayCode = extractDisplayFromConcept(immunization.vaccineCode);
-        if (displayCode) {
-          const key = JSON.stringify({ date, displayCode });
-          fillMaps(
-            displayMap,
-            key,
-            immunization,
-            refReplacementMap,
-            undefined,
-            assignMostDescriptiveStatus
-          );
-        } else {
-          danglingReferences.add(createRef(immunization));
-        }
-      }
+
+    const { cvxCode, ndcCode } = extractCodes(immunization.vaccineCode);
+    const display = extractDisplayFromConcept(immunization.vaccineCode);
+
+    const identifiers = [
+      ...(cvxCode ? [{ cvxCode }] : []),
+      ...(ndcCode ? [{ ndcCode }] : []),
+      ...(display ? [{ displayCode: display }] : []),
+    ];
+    const hasIdentifier = identifiers.length > 0;
+
+    if (!hasIdentifier) {
+      danglingReferences.add(createRef(immunization));
+      continue;
+    }
+    const getterKeys: string[] = [];
+    const setterKeys: string[] = [];
+
+    if (date) {
+      // keys that match a code + date together
+      setterKeys.push(...createKeysFromObjectArray({ date }, identifiers));
+      getterKeys.push(...createKeysFromObjectArray({ date }, identifiers));
+
+      // flagging the vaccine to indicate having a date
+      setterKeys.push(...createKeysFromObjectArrayAndBits(identifiers, [1]));
```

Comment:
I assume `[1]` mean something special, as well as `[0]` - and so on. Can we create constant for these that have name that conveys that meaning, please?

And/or create functions that are bound to that particular context, like:
- `createKeysFromObjectArrayForContextABC()`
- `createKeysFromObjectArrayForTheOtherSituation()`